src/fragments/lib/datastore/native_common/advanced-workflows.mdx

```diff
@@ -0,0 +1,207 @@
+## Custom Primary Keys
+
+By default, all DataStore models have an `id` field by default that is automatically populated on the client with a UUID v4, allowing DataStore to generate non-colliding globally unique identifiers in a scalable way.
+
+While UUIDs have desirable properties (they are large, non-sequential and opaque), there are times when a custom primary key is needed. For instance, to:
+
+- Have friendly/readable identifiers (surrogate/opaque vs. natural keys)
+- Define composite primary keys
+- Customize data partitioning to optimize for scale (especially important when planning to handle large amounts of data in short periods of time)
+- Selectively synchronize data to clients (e.g. by fields like `deviceId`, `userId` or similar)
+- Prioritize the sort order in which objects are returned by the sync queries
+- Make existing data consumable and syncable by DataStore
+
+### Example schemas using custom primary keys
+
+A schema with the typical `id` field looks like this:
+
+```graphql
+type Book @model {
+  id: ID!
+  title: String!
+  description: String
+}
+```
+
+With custom primary keys, you can provide a unique field name and/or value using the `@primaryKey` directive:
+
+<BlockSwitcher>
+<Block name="Custom Primary Key">
+
+```graphql
+type Book @model {
+  isbn: ID! @primaryKey
+  title: String!
+  description: String
+}
+```
+
+</Block>
+<Block name="Composite Key">
+
+```graphql
+type Book @model {
+  isbn: ID! @primaryKey(sortKeyFields: ["title"])
+  title: String!
+  description: String
+}
+```
+
+</Block>
+</BlockSwitcher>
+
+### Creating records with Custom Primary Keys:
+
+The following table explains when a value for the custom primary key must be provided:
+
+<table>
+<tbody>
+<tr>
+<th>Description</th>
+<th>Type</th>
+<th>Autopopulated with UUID</th>
+</tr>
+<tr>
+<td>Without @primaryKey</td>
+<td>
+
+```graphql
```

Comment:
@renebrandel these are newly added. Had to remove the table indentation to allow me to include the code snippets, sorry.

Reply:
LGTM